### ISSUE TITLE
Upgrading from 5.1 to 5.2 produces multiple errors in field labels.

### ISSUE THREAD

**Issue by J-Wick4** [orangehrm/orangehrm#1500]
I just upgraded, and most of the site is missing proper field labels.

![Screen Shot 2022-10-19 at 7 30 04 PM](https://user-images.githubusercontent.com/1954540/196829193-71acaf1e-7f25-4169-ac04-20d1e6773fc0.png)

Unpacked the 5.2 folder, ran the wizard, selected upgrade from 5.1, and had all great checkmarks system check.

**Comment by cholloSalvador** [comment]
Same issue.

**Comment by RajithaKumara** [comment]
Hi @J-Wick4 ,
Did you unpacked OrangeHRM 5.2 into old OrangeHRM 5.1 folder?
When upgrading it's recommended to backup and delete old version codebase and place new codebase in that location. Otherwise some files may conflicted. I guess here cache folder got conflicted with old cache folder.

There is a simple solution, can you run below command at the root of the orangehrm codebase;
```bash
php bin/console cache:clear
```

**Comment by J-Wick4** [comment]
Hey @RajithaKumara, Yes, that's how I did the upgrade, using the new unpacked folder. I'll give it another try tomorrow and run your command. Thank you.

**Comment by cholloSalvador** [comment]
> 

How to run command?

**Comment by RajithaKumara** [comment]
Hi @cholloSalvador ,
if you are the server administrator, login/ssh to the server. And locate the orangehrm application directory.
```sh
cd /path/to/orangehrm
php bin/console cache:clear
```

**Comment by cholloSalvador** [comment]
> 

sorry I dont have any idea. Can you share me a sample video? Thanks

**Comment by salmansiddiqui17** [comment]
I have the same issue. I tried cleaning the cache but it shows this error
<img width="988" alt="image" src="https://user-images.githubusercontent.com/33417633/197785851-974ddc3c-01f0-4dca-bbb5-a4423f90cd05.png">

**Comment by salmansiddiqui17** [comment]
I also tried a fresh installation but faced the same issue. I guess it's a bug.

**Comment by salmansiddiqui17** [comment]
I executed php bin/console cache:clear but still same issue

<img width="728" alt="image" src="https://user-images.githubusercontent.com/33417633/197790388-a5678fe4-34c3-4744-af7d-fd82032689cf.png">

**Comment by RajithaKumara** [comment]
Hi @salmansiddiqui17 ,
is it possible you to open developer tool network tab and provide more info like below screenshot;

![Screenshot from 2022-10-25 19-50-21](https://user-images.githubusercontent.com/25487187/197800304-714077dc-87a7-497f-b0b0-d3803313b1a7.png)

**Comment by salmansiddiqui17** [comment]
@RajithaKumara Response is empty but status code is 304 not modified
<img width="1672" alt="image" src="https://user-images.githubusercontent.com/33417633/198024700-7a8a5ccc-bb55-4ca4-b780-c8da36efdd61.png">
![Uploading image.png…]()

**Comment by Super-Chama** [comment]
Hi @salmansiddiqui17 It looks like the response is fine. Can you change the language to English US (if not already) then check and using the developer tools, please confirm any console logs are displayed.

1. Change Language to English US
    - Navigate to Admin -> Configuration -> Localization.
    - Change Language drop-down to "English (United States)".
    - Click Save.
    - Check if problem is resolved.
![image](https://user-images.githubusercontent.com/12880059/198199443-e161e6c2-1167-4506-a59a-997215b82a61.png)

2. Confirm No Error Logs
    - After page is fully loaded press `Ctrl + Shift + I` to Open browser developer tools.
    - Select `Console` tab
    - Check if any errors are present.
![image](https://user-images.githubusercontent.com/12880059/198199822-589abbc3-37ee-428a-b0b1-5f7f9053eeef.png)

**Comment by pbaranski** [comment]
Hi, having the same problem.
When loading in incognito first time it is ok. When reloading then bug appear. 
When local storage in browser is removed  an reload applied then site looks fine until next request.
This is server installation - locally it works fine. Guess some server dependencies.

**Comment by Super-Chama** [comment]
@pbaranski sorry to hear that. Language strings are cached in local storage from first request onwards. please check if your browser supports/allow local storage (by default this is enabled in chrome/firefox/safari) also additional information such as console logs and browser type,version is appreciated.

**Comment by pbaranski** [comment]
@Super-Chama 
Browsers are OK - tested on clean Chrome installation.
Looks like hit to local storage is somehow broken

@salmansiddiqui17 @J-Wick4  @cholloSalvador 
Quick fix for that problem is turning off redirection:
modify app file: `\src\plugins\orangehrmI18NPlugin\Controller\I18NMessagesController.php`
comment `if ` lines (add `//`) like below:
```
        // if (!$response->isNotModified($request)) {
            $response->setContent($this->getI18NService()->getTranslationMessagesAsJsonString($locale));
            $this->setCommonHeaders($response, 'application/json');
        // }

**Comment by salmansiddiqui17** [comment]
@pbaranski localization language is already English (US). Here is the SS of the Console

<img width="1673" alt="image" src="https://user-images.githubusercontent.com/33417633/198953944-332066b1-ebb6-4e2a-8bcd-f68c0b0ef458.png">

**Comment by salmansiddiqui17** [comment]
@pbaranski Thanks I'll try that.

**Comment by SupperDog** [comment]
> @Super-Chama Browsers are OK - tested on clean Chrome installation. Looks like hit to local storage is somehow broken
> 
> @salmansiddiqui17 @J-Wick4 @cholloSalvador Quick fix for that problem is turning off redirection: modify app file: `\src\plugins\orangehrmI18NPlugin\Controller\I18NMessagesController.php` comment `if ` lines (add `//`) like below:
> 
> ```
>         // if (!$response->isNotModified($request)) {
>             $response->setContent($this->getI18NService()->getTranslationMessagesAsJsonString($locale));
>             $this->setCommonHeaders($response, 'application/json');
>         // }
> ```

This fix doesn't work for me.

**Comment by kdy294** [comment]
Yeah I’m experiencing the same issue. It was a fresh install and everything was good to go. Just labels not appearing correctly or button labels.

**Comment by mleader1** [comment]
It turned out the value stored in localStorage in browsers are having some issues @SupperDog 

I solve it temporarily using the code below by appending into /web/dist/js/app.js

`
function languageFix() {
window.localStorage.setItem('"/RveX7hH7+0PZ5+FUVr/pX4WeN/w5Dn7zeeOwHWp+tA="', window.localStorage.getItem('W/"/RveX7hH7+0PZ5+FUVr/pX4WeN/w5Dn7zeeOwHWp+tA="'));
window.localStorage.setItem('core/i18n/messages','"/RveX7hH7+0PZ5+FUVr/pX4WeN/w5Dn7zeeOwHWp+tA="');
}

setTimeout(languageFix, 500);
`

It works neatly so far.   I'm first timer in using OrageHRM, can't wait for for the proper fix and more feature releases!

Cheers

**Comment by joelicatajr** [comment]
> > @Super-Chama Browsers are OK - tested on clean Chrome installation. Looks like hit to local storage is somehow broken
> > @salmansiddiqui17 @J-Wick4 @cholloSalvador Quick fix for that problem is turning off redirection: modify app file: `\src\plugins\orangehrmI18NPlugin\Controller\I18NMessagesController.php` comment `if ` lines (add `//`) like below:
> > ```
> >         // if (!$response->isNotModified($request)) {
> >             $response->setContent($this->getI18NService()->getTranslationMessagesAsJsonString($locale));
> >             $this->setCommonHeaders($response, 'application/json');
> >         // }
> > ```
> 
> This fix doesn't work for me.

I am experiencing the same issue. New clean installation. This fix also doesn't work for me.

Thanks for your help with this one.

**Comment by joelicatajr** [comment]
Just a FYI for anyone else experiencing this problem. I wiped my install and deployed with 5.1 instead of 5.2 to resolve my issue since there doesn't appear to be a conclusive working fix I could find for 5.2.

**Comment by RajithaKumara** [comment]
This issue was fixed in OrangeHRM 5.3. Get the latest build [here](https://sourceforge.net/projects/orangehrm/).

**Comment by J-Wick4** [comment]
Everything is working for me now, thanks.
